app/models/proposal.rb

```diff
@@ -47,7 +47,7 @@ def on_rejected_entry(prev_state, event)
   def restart
     # Note that none of the state machine's history is stored
     self.cart.api_tokens.update_all(expires_at: Time.now)
-    self.cart.approver_approvals.each do |approval|
+    self.cart.approver_approvals.where.not(status: 'approved').each do |approval|
```

Comment:
I'd avoid "restart"ing altogether if you want to keep it in and renaming this completely if not. Perhaps you want a "edited" message that causes the proposal to stay in the pending state, but triggers emails. Certainly don't want to `deliver_new_cart_emails` as this no longer resembles a new cart